### ISSUE TITLE
Banner for every destination file

### ISSUE THREAD

**Issue by piuccio** [gruntjs/grunt-contrib-concat#87]
I'd like to add a banner on top of my concatenated files that tells me which filepath were concatenated.

``` js
{
  concat: {
    dist: {
      "out1.js": ["src/one/file1.js", "src/one/file2.js"],
      "out2.js": ["src/two/another1.js", "src/two/another2.js"]
    }
  }
}
```

What I'd like to have as output is 
- out1.js

``` js
/**
 * This file is the concatenation of
 * src/one/file1.js
 * src/one/file2.js
 */
// code of file1.js ...
// code of file2.js ...
```

Similar output for out2.js

grunt-contrib-concat has two options that seem to do the job, but
- `banner`: this is generated once for all destination files, it has no knowledge of individual files
- `process`: this is done for every source file, not for the destination file

Using `process` I'd get

``` js
// I can add some code here
// code of file1.js ...
// and I can add some code here
// code of file2.js ...
```

What about adding another option to process the destination file?

Say for instance changing [line 38 and following](https://github.com/gruntjs/grunt-contrib-concat/blob/master/tasks/concat.js#L38) to be

``` js
// Instead of 
/**
var src = banner + f.src.filter(
    ...
).join(options.separator) + footer;
**/
var src = banner + options.processDestination(
    f.src.filter(/* ... */).join(options.separator), f.dest
) + footer;
```

So basically adding `processDestination` as a function with the same signature of `process` but working on destination file.

What do you think?
